### PR TITLE
Update `distributions.independent`

### PR DESCRIPTION
**Pull request by ganow**
I reflected the points I got in the previous pull request (#6324).

### CODE REVIEW

**Comment by ganow** [comment]
> LGTM except for a docstring.

I added what the `Independent.covariance` returns (a06e24f), and change the single quotation which was used for the docstring of `_kl_independent_independent(dist1, dist2)` to double (0129ca5). I guess this is what you've expected. If not, may I ask you which part of the docstring doesn't fit?

File: chainer/distributions/independent.py [hunk 1]
```diff
@@ -60,10 +58,12 @@ def batch_shape(self):
     def event_shape(self):
         return self.__event_shape
 
-    @property
+    @cache.cached_property
     def covariance(self):
-        '''Returns the covariance of the distribution based on the original
-        i.i.d. distribution. By definition, the covariance of the new
+        """ The covariance of the distribution based on the original
+        i.i.d. distribution.
```
Comment (toslunar) [inline chainer/distributions/independent.py]: Could you make the summary shorter to follow H405?

> [H405] Multi line docstrings should start with a one line summary followed by an empty line.

https://docs.openstack.org/hacking/latest/user/hacking.html#docstrings

In the previous review, I was going to include a comment like this on the line.  I'm sorry that I didn't.

Comment (ganow) [reply chainer/distributions/independent.py]: I missed that, sorry. Thank you for pointing out. I fixed them in cbc0e2b

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
The Jenkins failures are unrelated to the PR.

**Pull request merged**